Source/BestProjectEver/Public/AI/Services/BPE_BTService_GetDetectedPlayer.h

```diff
@@ -0,0 +1,56 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "BehaviorTree/BTService.h"
+#include "BPE_BTService_GetDetectedPlayer.generated.h" 
+
+class UAISense;
+class ABPE_Enemy;
+/**
+ * 
+ */
+UCLASS()
+class BESTPROJECTEVER_API UBPE_BTService_GetDetectedTarget : public UBTService
+{
+	GENERATED_BODY()
+	
+protected:
+
+	/** target detected by controller */
+	UPROPERTY(EditAnywhere, Category="Blackboard")
+	FName TargetReferenceName;
+
+	/** target detected by controller location */
+	UPROPERTY(EditAnywhere, Category="Blackboard")
+	FName TargetLocationName;
+
+	TObjectPtr<AAIController> AIController;
```

Comment:
Pointers are dangerous if not used with UPROPERTYs. When you have a UObject-type pointer and you don't need any UPROPERTY specifier, you should use `UPROPERTY(Transient)` to avoid having them garbage collected. (This is especially true for non-actors, such as your BlackboardComponent, but still it's better to do it with all UObject types)

This is a change that needs to be done on many classes in this review :)

Reply:
Done

Reply:
:)))